eslint.config.mjs

```diff
@@ -0,0 +1,131 @@
+import { fixupPluginRules } from "@eslint/compat";
+import typescriptEslint from "@typescript-eslint/eslint-plugin";
+import importPlugin from "eslint-plugin-import";
+import globals from "globals";
+import tsParser from "@typescript-eslint/parser";
+import nextPlugin from "@next/eslint-plugin-next";
+import reactHooks from "eslint-plugin-react-hooks";
+
+export default [
+  {
+    ignores: [
+      "**/dist",
+      "**/build",
+      "**/node_modules",
+      "**/*.config.js",
+      "**/*.config.mjs",
+      "**/eslint.config.mjs",
+      "**/vite.config.ts",
+      "tailwind.config.js",
+      "postcss.config.js",
+      "**/*.js",
+      ".storybook/preview.ts",
+      ".storybook/main.ts",
+    ],
+  },
+  {
+    files: ["**/*.{ts,tsx}"],
+    plugins: {
+      "@typescript-eslint": fixupPluginRules(typescriptEslint),
+      "@next/next": nextPlugin,
+      import: fixupPluginRules(importPlugin),
+      "react-hooks": reactHooks,
+    },
+
+    languageOptions: {
+      globals: {
+        ...globals.browser,
+        ...globals.es2021,
```

Comment:
suggestion: As we're using `esnext` in `tsconfig`, could we use the most modern version here i.e. `es20205`?